apim/values.yaml

```diff
@@ -363,19 +380,24 @@ gateway:
     maxReplicas: 6
     targetAverageUtilization: 50
     targetMemoryAverageUtilization: 80
+  # This is usefull in case of dynamic rewriting
+  entrypoint: /gateway/
   ingress:
     enabled: true
-    path: /gateway
+    path: /gateway/(.*)
     # Used to create an Ingress record.
     hosts:
       - apim.example.com
     annotations:
       kubernetes.io/ingress.class: nginx
       nginx.ingress.kubernetes.io/ssl-redirect: "false"
       nginx.ingress.kubernetes.io/enable-rewrite-log: "true"
-      kubernetes.io/app-root: /gateway
-      kubernetes.io/rewrite-target: /gateway
-      # ingress.kubernetes.io/configuration-snippet: "etag on;\nproxy_pass_header ETag;\nproxy_set_header if-match \"\";\n"
+      # App-root can be used for convenience purpose if
+      # ingress hosts are only used for the gateway on the path '/'
+      # (Value should be similar to ingress.path)
+      # nginx.ingress.kubernetes.io/app-root: /gateway/
+      nginx.ingress.kubernetes.io/rewrite-target: /$1
```

Comment:
So, as far as I understand, it means that when creating API and defining their context-path we don't have to put the `/gateway` anymore, am I right?

And, in that case, it means that the one who was using previous version of the chart will have to update all his API's context-path to make it works.

Please confirm.

Reply:
You are right, and you are also right this will break things  up. This slipped through my review, as this is something we needed in our use case, but the goal of this PR was to keep changes to a minimum for current users.